### PR TITLE
Update user and developer guide to reflect our current implementations

### PR DESCRIPTION
**Pull request by kangzongxian**
Added implementation of improved `find` command
Added How to use User Guide in User Guide
Fix styling issues

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -235,9 +243,9 @@ Common mistakes:
 
 ### Clearing all entries : `clear`
 
-This command clears all entries from the guest book. 
+This command clears all entries from GuestBook. 
 
-This command is typically used as a hard reset of the entire guest book. Do note that this command 
+This command is typically used as a hard reset of GuestBook. Do note that this command
```
Comment (bryanhce) [inline docs/UserGuide.md]: Perhaps for this sentence, we might want to write it as "a hard reset of all guest data in GuestBook", for more clarity?

**Review by bryanhce — COMMENTED** [review]
nice work!

File: docs/UserGuide.md (hunk 1 above)
Comment (kangzongxian) [reply docs/UserGuide.md]: Agreed!

**Review by bryanhce — APPROVED** [review]
LGTM

**Pull request merged**